app/models/upload.rb

```diff
@@ -361,13 +356,7 @@ def calculate_dominant_color!(local_path = nil)
         if local?
           Discourse.store.path_for(self)
         else
-          begin
-            Discourse.store.download(self)&.path
-          rescue OpenURI::HTTPError => e
```

Comment:
This was insufficient, because the collaborators of `FileStore#download` can raise other exceptions, like the ones coming from `FinalDestination::SSRFDetector`.